tools/azsdk-cli/Azure.Sdk.Tools.Cli/Helpers/SpecGenSdkConfigHelper.cs

```diff
@@ -93,42 +98,45 @@ public async Task<T> GetConfigValueFromRepoAsync<T>(string repositoryRoot, strin
             }
         }
 
-        // Get build configuration (either command or script path)
-        public async Task<(BuildConfigType type, string value)> GetBuildConfigurationAsync(string repositoryRoot)
+        // Get configuration for a specific type (either command or script path)
+        public async Task<(SpecGenSdkConfigContentType type, string value)> GetConfigurationAsync(string repositoryRoot, SpecGenSdkConfigType configType)
         {
+            var (commandPath, scriptPath) = GetConfigPaths(configType);
+            
             // Try command first
             try
             {
-                var command = await GetConfigValueFromRepoAsync<string>(repositoryRoot, BuildCommandJsonPath);
+                var command = await GetConfigValueFromRepoAsync<string>(repositoryRoot, commandPath);
                 if (!string.IsNullOrEmpty(command))
                 {
-                    _logger.LogDebug("Found build command configuration");
-                    return (BuildConfigType.Command, command);
+                    _logger.LogDebug("Found {ConfigType} command configuration", configType);
+                    return (SpecGenSdkConfigContentType.Command, command);
                 }
             }
             catch (InvalidOperationException)
             {
                 // Command not found, continue to try path
-                _logger.LogDebug("No build command found, trying script path");
+                _logger.LogDebug("No {ConfigType} command found, trying script path", configType);
             }
 
             // Try path
             try
             {
-                var path = await GetConfigValueFromRepoAsync<string>(repositoryRoot, BuildScriptPathJsonPath);
+                var path = await GetConfigValueFromRepoAsync<string>(repositoryRoot, scriptPath);
```

Comment:
Is there a good reason to treat cmd vs script differently? Why not all be cmd and folks have to prefix with the correct tool. (i.e `pwsh script.ps1`)? While we might move away from this existing implementation I wanted to understand if there is a good reason to have these be forked in the code and config.

Reply:
The main reason is the parameter mapping logic is different when preparing the command call vs the script call.
for example, 
script is configured as `build-sdk.ps1`, we need to prepare the call as `pwsh build-sdk.ps1 -packagePath {value}`;
command is configured like [this sample](https://github.com/Azure/azure-sdk-for-java/blob/31ef7cd8cccd2e0f93532e07715aaf186f5dac5b/eng/swagger_to_sdk_config.json#L30), we need to replace the {packagePath} by the real value.

Or we need to align the script config format with same inline parameter name and parameter value. i.e. `build-sdk.ps1 -packagePath {packagePath}`.